### PR TITLE
Configure clang-tidy to uniformise the style in naming classes, class data members and variables

### LINKED ISSUE #494: Configure clang-tidy to uniformise the style in naming classes, class data members and variables
- [x] Use `CamelCase` for class name
- [x] Use `lower_case` for data members with a trailing underscore

### PR DESCRIPTION
**Pull request by boulderdaze**
Closes #494 
 - Use `CamelCase` for class name
 - Use `lower_case` for data members with a trailing underscore
 - Use `UPPER_CASE` for constants

### CODE REVIEW

File: .clang-tidy [hunk 1]
```diff
@@ -54,6 +54,20 @@ CheckOptions:
...
+  - key:             readability-identifier-naming.ClassCase
+    value:           CamelCase
+  - key:             readability-identifier-naming.ClassIgnoredRegexp
+    value:           'is_error_condition_enum'
+  - key:             readability-identifier-naming.MemberCase
+    value:           lower_case
+  - key:             readability-identifier-naming.MemberSuffix
+    value:           _
+  - key:             readability-identifier-naming.MemberIgnoredRegexp
+    value:           '^._$|^k0_._$|^kinf_._$|^Fc_$|^niLU_.*$|[LU]ij_.*$'
```
Comment (boulderdaze) [inline .clang-tidy:66]: Rethinking this. Maybe we shouldn't require certain case, so we don't need to list exception cases for math parameters. @mattldawson, @K20shores, @mwaxmonsky, what do you think? 
```suggestion
  - key:             readability-identifier-naming.MemberCase
    value:           aNy_CasE
  - key:             readability-identifier-naming.MemberSuffix
    value:           _
```

Comment (K20shores) [reply .clang-tidy:66]: That seems reasonable; we do work in a math field after all and not every variable can easily be named with camel case

**Review by K20shores — APPROVED** [review]

File: .clang-tidy (hunk 1 above)
Comment (boulderdaze) [reply .clang-tidy]: I agree with you. I updated it to allow any case for variable names

**Review by mattldawson — APPROVED** [review]

**Review by mwaxmonsky — APPROVED** [review]
Looks great @boulderdaze!

**Pull request merged**